web/client/components/geostory/layouts/sections/enhancers/immersiveBackgroundManager.jsx

```diff
@@ -53,11 +53,18 @@ const createBackgroundIdStream = (intersection$) =>
  * from `contents` property
  */
 export const backgroundProp = withProps(
-    ({ backgroundId, contents = [] }) => ({
-        background: get(getContentIndex(contents, backgroundId) || 0, 'background') || {
-            type: 'none'
-        }
-    }));
+    ({ id, contents = [] }) => {
+        const contextIndex = getContentIndex(contents, id);
+        return {
+            background: {
+                ...get(contextIndex || 0, 'background') || {
+                    type: 'none'
+                }
+            },
+            contentId: contextIndex && contextIndex.id,
+            sectionId: id
+        };
```

Comment:
I think we should not remove the backgroundId prop because it used in the stream for immersive layout to select the current background in view.
It seems also getContentIndex returns a content object instead of an index so I propose to change it to getContentInView.

example
```js
({ id, backgroundId, contents = [] }) => {
    const contentInView = getContentInView(contents, backgroundId) || {};
    return {
        background: get(contentInView, 'background') || { type: 'none' },
        contentId: contentInView && contentInView.id,
        sectionId: id
    };
}
```